src/key.js

```diff
@@ -623,9 +625,38 @@ async function mergeSignatures(source, dest, attr, checkFn) {
   }
 }
 
-// TODO
-Key.prototype.revoke = function() {
-
+/**
+ * Revokes a Key
+ * @param {Integer} reasonFlag optional reason for revocation
+ * @param {Integer} reasonString optional additional description string
+ * @async
+ */
+Key.prototype.revoke = async function(reasonFlag, reasonString){
```

Comment:
The reasonString should be used more than reasonFlag, so exchange the params position may be good.

```js
Key.prototype.revoke = async function(reasonString, reasonFlag)
```